pki/denylist.go

```diff
@@ -224,10 +203,19 @@ func (b *denylistImpl) Update() error {
 	// Log when the denylist is updated
 	logger().Debug("Denylist updated successfully")
 
+	// Notify all subscribers synchronously
+	for _, sub := range b.subscribers {
+		sub()
```

Comment:
can `conn.disconnect()` block when the other side is a slow responder (closing TCP involves sending packets to thw other side)? Since subscribers are called synchronously?